docs/serviceregistry/manuals/temperature-http-demo.md

```diff
@@ -0,0 +1,846 @@
+- **addresses:** This is typically an IP address or a hostname for the system. In case of the _temperature-provider2_, we want to provide both. For IP address, we are using 192.168.56.116, and the hostname is tp2.greenhouse.com.
+- **device name:** This can be left blank, but we will set it to the name of our device: thermometer2.
+
+It is not necessary to specify the system name explicitly, because the Service Registry extracts it from the authorization header.
+
+Based on the above, the request looks like this:
+
+~~~
+curl -X 'POST' \
+  'http://localhost:8443/serviceregistry/system-discovery/register' \
+  -H 'accept: application/json' \
+  -H 'Authorization: Bearer SYSTEM//temperature-provider2' \
+  -H 'Content-Type: application/json' \
+  -d '{
+  "metadata": {
+    "scales": ["Kelvin", "Celsius"],
+    "location": {"side": "North", "block": 2},
+    "indoor": true
+  },
+  "version": "",
+  "addresses": [
+    "192.168.56.116",
+    "tp2.greenhouse.com"
+  ],
+  "deviceName": "thermometer2"
+}'
+~~~
+
+We receive the following response:
+
+~~~
+{
+  "name": "temperature-provider2",
+  "metadata": {
+    "scales": [
+      "Kelvin",
+      "Celsius"
+    ],
+    "location": {
+      "side": "North",
+      "block": 2
+    },
+    "indoor": true
+  },
+  "version": "1.0.0",
+  "addresses": [
+    {
+      "type": "IPV4",
+      "address": "192.168.56.116"
+    },
+    {
+      "type": "HOSTNAME",
+      "address": "tp2.greenhouse.com"
+    }
+  ],
+  "device": {
+    "name": "thermometer2",
+    "metadata": {
+      "scales": [
+        "Kelvin",
+        "Celsius"
+      ],
+      "max-temperature": {
+        "Kelvin": 310,
+        "Celsius": 40
+      },
+      "min-temperature": {
+        "Kelvin": 260,
+        "Celsius": -10
+      }
+    },
+    "addresses": [
+      {
+        "type": "MAC",
+        "address": "81:ef:1a:44:7a:f5"
+      }
+    ],
+    "createdAt": "2024-11-04T01:53:02Z",
+    "updatedAt": "2024-11-04T01:53:02Z"
+  },
+  "createdAt": "2024-11-08T10:21:10.950683800Z",
+  "updatedAt": "2024-11-08T10:21:10.950683800Z"
+}
+~~~
+
+### Step 4: Register Service Instances
+
+The _temperature-provider2_ system that we have just registered provides three services in the Local Cloud:
+- **Kelvin info:** provides temperature information using the Kelvin scale,
+- **Celsius info:** provides temperature information using the Celsius scale,
+- **alert service:** sends an alert if the temperature is extreme (by default, these thresholds are 10 and 25 Celsius, but the consumer can overwrite them).
+
+We have to registrer these services one by one.
+
+**1. Kelvin info:** 
+We have to provide the following information:
+- **service definition name:** We have to use the name of one of the existing service definitions stored in the Local Cloud. In this example, this is kelvin-info.
```

Comment:
This is not true. If the service definition not exist, the registration process will create them.